NODE-885: Remove transport layer

aakoshh opened the pull request:
### Overview
Removes transport layer.

### Which JIRA ticket does this PR relate to?
https://casperlabs.atlassian.net/browse/NODE-885

### Complete this checklist before you submit this PR
- [ ] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [CasperLabs coding standards](https://casperlabs.atlassian.net/wiki/spaces/EN/pages/16842753/Coding+Standards).
- [ ] If this PR adds a new feature, it includes tests related to this feature.
- [x] You assigned one person to review this PR.
- [x] Your GitHub account is linked with our [Drone CI](https://drone-auto.casperlabs.io/) system. This is necessary to run tests on this PR.

### Notes
This leaves the nodes unable to validate the Genesis candidate. That will be rectified in https://github.com/CasperLabs/CasperLabs/pull/1137

goral09 commented:
Finally.

goral09 commented on casper/src/main/scala/io/casperlabs/casper/util/DagOperations.scala:
```diff
@@ -353,6 +354,11 @@ object DagOperations {
       .map(_.blockHash)
   }
 
+  private implicit val blockMetadataEq = new cats.kernel.Eq[BlockMetadata] {
```
I think we already have these instances somewhere. Can you put it in `io.casperlabs.casper.util.implicits` ?

aakoshh replied:
I must have deleted it. I wasn't sure if it's going to be needed moving forward since Igor removed it completely, but I'll put it there, easier to find to delete again.

goral09 commented on casper/src/test/scala/io/casperlabs/casper/ManyValidatorsTest.scala:
```diff
@@ -80,10 +80,10 @@ class ManyValidatorsTest extends FlatSpec with Matchers with BlockGenerator with
-                 casperRef,
                  logEff,
+                 casperRef,
                  finalityDetectorEffect,
```
Are these still `*Effect` ? if the effect i `Task` ?

aakoshh replied:
I thought you can choose your effect to be whatever you want. Be it Task, IO, or EitherT[Task, String, A]

goral09 commented on casper/src/test/scala/io/casperlabs/casper/helper/HashSetCasperTestNode.scala:
```diff
@@ -198,70 +194,10 @@ trait HashSetCasperTestNodeFactory {
 
 object HashSetCasperTestNode {
-  type Effect[A]                        = EitherT[Task, CommError, A]
+  type Effect[A]                        = Task[A]
```
Would it be too much to ask to actually replace `Effect with `Task` everywhere? :)

aakoshh replied:
Maybe it's easier for IntelliJ. I can try to do a string replace.

goral09 commented on node/src/main/scala/io/casperlabs/node/package.scala:
```diff
@@ -9,26 +9,17 @@ import cats.temp.par.Par
   /** Final Effect + helper methods */
-  type CommErrT[F[_], A] = EitherT[F, CommError, A]
-  type Effect[A]         = CommErrT[Task, A]
+  type Effect[A] = Task[A]
```
I am seeing it for the second time already.

aakoshh commented:
That `LastApprovedBlock` was only dealing with the `ApprovedBlock` of the old stuff, the newer `GenesisCandidate` was never persisted. What I did with the newer Genesis ceremony was that when I get a `consensus.Block` I converted it to `protocol.BlockMessage` using `LegacyConversions` and passed it to `BlockApproverProtocol` for validation. 

But since @birchmd doesn't want to keep the validation I thought I'll save myself the trouble of converting `BlockApproverProtocol` to the `consensus` types and removed the whole lot. So it's not able to validate a candidate as it does on `dev`, but the plan is to add the ChainSpec, create a Genesis block from that, and just compare hashes.

goral09 commented:
So there's no Genesis process in this PR?

aakoshh commented:
The `approveGenesis` mode is [disabled](https://github.com/CasperLabs/CasperLabs/pull/1143/files#diff-8658c5ba50c0f26078f9c07220038b0cR394).

aakoshh commented:
bors r+

aakoshh commented:
bors r+

aakoshh commented:
bors r+

aakoshh commented:
bors r+

aakoshh commented:
bors r+
